server/apps/microservices/src/processors/video-transcode.processor.ts

```diff
@@ -72,10 +73,45 @@ export class VideoTranscodeProcessor {
   async runVideoEncode(asset: AssetEntity, savedEncodedPath: string): Promise<void> {
     const config = await this.systemConfigService.getConfig();
 
-    if (config.ffmpeg.transcode == TranscodePreset.ALL) {
+    const transcode = await this.needsTranscoding(asset, config.ffmpeg);
+    if (transcode) {
+      //TODO: If video or audio are already the correct format, don't re-encode, copy the stream
       return this.runFFMPEGPipeLine(asset, savedEncodedPath);
     }
+  }
+
+  async needsTranscoding(asset: AssetEntity, ffmpegConfig: SystemConfigFFmpegDto): Promise<boolean> {
+    switch (ffmpegConfig.transcode) {
+      case TranscodePreset.ALL:
+        return true;
+
+      case TranscodePreset.REQUIRED:
+        {
```

Comment:
I think these extra brackets around the case body should be unnecessary. Did you run the formatter?

Reply:
They are necessary, because `videoStream` is used twice and the switch statement doesn't create [lexical scopes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#lexical_scoping)

Reply:
> Did you run the formatter?

Yes. In my implementation these braces are necessary to limit the scope of `videoStream` variable. Without them there is an error `Cannot redeclare block-scoped variable 'videoStream'`